src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4232,6 +4232,10 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <ItemGroup Condition="'$(PublishSingleFile)' == 'true'">
       <_ClickOnceFiles Include="$(PublishedSingleFilePath)"/>
       <_ClickOnceFiles Include="@(_FilesExcludedFromBundle)"/>
+
+      <!-- Include file association icons from Content as loose files -->
+      <_FileAssociationIcons Include="%(FileAssociation.DefaultIcon)"/>
+      <_ClickOnceFiles Include="@(ContentWithTargetPath)" Condition="'%(Identity)'=='@(_FileAssociationIcons)'"/>
```

Comment:
I don't think this will work if there's more than one `FileAssociation.DefaultIcon`. Have you tested that case?

Reply:
Good point, I had not. I just tried it and it seems to work. 

![image](https://user-images.githubusercontent.com/7729842/122481091-efc07e00-cf82-11eb-9b39-6f69b73ed685.png)

Reply:
> Is this the one y'all would like to have for 16.11, or is 17.0 (our current main branch) ok?

Current plan is to fix this for Dev17. We haven't had customers report this corner case failure. If we need to get this into 16.11 at this point, I suppose we need to go through shiproom for approval?

Reply:
I didn't understand how the batching would work as written and I spent a bit of time to build my understanding.

This expression has an unqualified metadata reference `%(Identity)`, so it applies the batching rules to _all item lists in the expression_. Since `_FileAssociationIcons` and `ContentWithTargetPath` both have the `.ico` files as entries, this works fine.

(Someday I will internalize this, but it's not today. And it wasn't two years ago either: #4429)

I prefer to avoid this behavior when possible (because it's so confusing!). How would you feel about

```suggestion
      <_ClickOnceFiles Include="@(ContentWithTargetPath)" Condition="'%(ContentWithTargetPath.Identity)'=='@(_FileAssociationIcons)'"/>
```

?

Reply:
Thanks makes sense. Essentially I want to iterate through items in ContentWithTargetPath item list and add the item to _ClickOnceFiles only if it exists in _FileAssociationIcons item list.
Is there a better way to express this in msbuild?

Reply:
@rainersigwald 
Just want to make sure the change I have is the correct way to express the intent. If so, I will go ahead with the merge.

Reply:
Ah, sorry. Yes, this is fine.